coding-test/minjae/BackJoon/7단계-문자열/그룹단어체커.js

```diff
@@ -0,0 +1,31 @@
+const readline = require("readline"); 
+const rl = readline.createInterface({ 
+    input: process.stdin, 
+    output: process.stdout, 
+}); 
+
+const input = [];
+
+rl.on("line", function (line) {
+    input.push(line); 
+}).on("close", function () { 
+    const testCase = parseInt(input.shift());
+
+    let count = 0;
+
+    for (let i = 0; i < testCase; i++) { 
+        const result = [];
+        const word = input[i].split("");
+
+        let checkGroupWord = true; 
+
+        word.map(el => {
+            if (result.indexOf(el) === -1) result.push(el);
+            else {
+                if (result.indexOf(el) !== result.length - 1) checkGroupWord = false; 
```

Comment:
indexOf 같이 인덱스를 구하는 알고리즘은 기본적으로 O(n)의 시간 복잡도가 소요됩니다.
아래처럼 여러번 사용하게 될 경우 그 만큼 시간이 소요되니 변수에 담고 그것을 사용해주는 것이 좋습니다.
```suggestion
            const idx = result.indexOf(el);
            if (idx === -1) result.push(el);
            else {
                if (idx !== result.length - 1) checkGroupWord = false; 
```